tap_core/src/tap_receipt/checks.rs

```diff
@@ -70,6 +78,58 @@ impl Check for TimestampCheck {
     }
 }
 
+pub struct BatchTimestampCheck(pub u64);
+
+impl CheckBatch for BatchTimestampCheck {
+    fn check_batch(
+        &self,
+        receipts: Vec<ReceiptWithState<Checking>>,
+    ) -> (
+        Vec<ReceiptWithState<Checking>>,
+        Vec<ReceiptWithState<Failed>>,
+    ) {
+        let (mut checking, mut failed) = (vec![], vec![]);
+        for receipt in receipts.into_iter() {
+            let receipt_timestamp_ns = receipt.signed_receipt().message.timestamp_ns;
+            let min_timestamp_ns = self.0;
+            if receipt_timestamp_ns >= min_timestamp_ns {
```

Comment:
```suggestion
            if receipt_timestamp_ns > min_timestamp_ns {
```

Reply:
done

Reply:
Actually was correct but needs a docstring explaining what timestamp check it does

Reply:
just added the docstring